### PR TITLE
Make `-- { echo }` hint preserve leading comments

### PR DESCRIPTION
**Pull request by akuzm**
Changelog category (leave one):
- Not for changelog (changelog entry is not required)

Extracted from https://github.com/ClickHouse/ClickHouse/pull/19022

### CODE REVIEW

**Comment by akuzm** [comment]
Fuzzer https://github.com/ClickHouse/ClickHouse/issues/19251

File: src/Parsers/parseQuery.cpp [hunk 1]
```diff
@@ -241,70 +250,90 @@ ASTPtr tryParseQuery(
...
 
-    if (!(insert && insert->data))
+    // If parsed query ends at data for insertion. Data for insertion could be
+    // in any format and not necessary be lexical correct, so we can't perform
+    // most of the checks.
+    if (insert && insert->data)
     {
-        /// Lexical error
-        if (last_token.isError())
+        if (!parse_res)
```
Comment (tavplubix) [inline src/Parsers/parseQuery.cpp:272]: @akuzm, it's always false

Comment (akuzm) [reply src/Parsers/parseQuery.cpp:272]: Oh.... So how does it manifest, we display bad messages for broken inserts? I have to add some test for this.

Comment (akuzm) [reply src/Parsers/parseQuery.cpp:272]: Looks like we can just remove it: https://github.com/ClickHouse/ClickHouse/pull/19919

**Pull request merged**